rllib/algorithms/algorithm_config.py

```diff
@@ -664,15 +660,8 @@ def update_from_dict(
                 # Resolve possible classpath.
                 value = deserialize_type(value)
                 self.rollouts(sample_collector=value)
-            # If config key matches a property, just set it, otherwise, warn and set.
+            # Set the property named `key` to `value`.
             else:
-                if not hasattr(self, key) and log_once(
-                    "unknown_property_in_algo_config"
-                ):
-                    logger.warning(
-                        f"Cannot create {type(self).__name__} from given "
-                        f"`config_dict`! Property {key} not supported."
-                    )
```

Comment:
This appears separate from the other changes. I think and I think it's a very useful warning.
We always had this usability issue of where users create configs that have keys with typos and other random stuff and we would be silent about it. That's one reason why we created the AlgorithmConfig after all, for proper validation. With this change, we offer a workflow that works against this, where folks can update with malformed dicts and not be warned about it, right?

Reply:
This is very true! However, tune seems to add its own keys to the config (when running through a `Tuner`), so tune does not care about our convention :) and messes up this thinking. Hence, whenever you (correctly!) use a AlgorithmConfig object, but then pass it into `tune.Tuner` you will get this warning.

Maybe we can filter out those keys/properties added by tune?